homeassistant/components/epion/config_flow.py

```diff
@@ -0,0 +1,70 @@
+"""Config flow for Epion."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from epion import Epion
+from requests.exceptions import ConnectTimeout, HTTPError
+from voluptuous import Required, Schema
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_API_KEY
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EpionConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Epion."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+        if user_input is not None:
+            try:
+                key_valid = await self.hass.async_add_executor_job(
+                    self._check_api_key, user_input[CONF_API_KEY]
+                )
+                if key_valid:
+                    return self.async_create_entry(
+                        title="Epion integration",
+                        data={CONF_API_KEY: user_input[CONF_API_KEY]},
+                    )
+                else:
+                    errors["base"] = "invalid_api_key"
```

Comment:
Only have stuff in the try block if it can raise

Reply:
Updated